src/components/Filters/SchemaSieve.ts

```diff
@@ -142,3 +149,125 @@ export const getOperators = (schema: JSONSchema6, field: string) => {
 		label: value.getLabel(subSchema),
 	}));
 };
+
+// Reduces a multi level schema to a single level
+export const flattenSchema = (
+	schema: JSONSchema6,
+	delimiter: string = DELIMITER,
+	parentKey: string = '',
+	accumulator?: JSONSchema6,
+): JSONSchema6 => {
+	if (!accumulator) {
+		accumulator = cloneDeep(schema);
+
+		if (schema.properties) {
+			accumulator.properties = {};
+		}
+	}
+
+	if (schema.anyOf) {
+		accumulator.anyOf = schema.anyOf.map(subSchema =>
+			flattenSchema(subSchema, delimiter, parentKey),
+		);
+	}
+
+	forEach(schema.properties, (value: JSONSchema6, key) => {
+		// Redeclare accumulator as TS doesn't know if the value is defined or not
+		// https://github.com/Microsoft/TypeScript/issues/15631
+		const acc = accumulator!;
+
+		const newKey = parentKey + DELIMITER + key;
+		// If the value is not an object type or
+		// If the value is a key value pair style object, we can skip it
+		if (
+			value.type !== 'object' ||
+			(findKey(value.properties, { description: 'key' }) &&
+				findKey(value.properties, { description: 'value' }))
+		) {
+			// Leave the data intact if we're not flattening it
+			if (parentKey) {
+				return (acc.properties![newKey] = defaults(value, { title: key }));
+			}
+
+			return (acc.properties![key] = value);
+		}
+
+		flattenSchema(value, delimiter, newKey, acc);
+	});
+
+	return accumulator;
+};
+
+// Restores a schema that has been flattened with `flattenSchema()`
+export const hydrateSchema = (
```

Comment:
Hydrate doesn't sound like the opposite of flatten, it sounds like it's deserializing data. `unflatten`?

Reply:
Good idea